internal/cluster/common.go

```diff
@@ -84,6 +85,30 @@ func updateClusterProgress(log logrus.FieldLogger, db *gorm.DB, clusterId strfmt
 	return cluster, nil
 }
 
+func updateLogsProgress(log logrus.FieldLogger, db *gorm.DB, clusterId strfmt.UUID, srcStatus string,
+	progress string, extra ...interface{}) (*common.Cluster, error) {
+	var cluster *common.Cluster
+	var err error
+
+	log.Infof("DEBUG => update log prgress on cluster %s to progress %s", clusterId, progress)
+
+	switch progress {
+	case string(models.LogsStateRequested):
+		extra = append(append(append(make([]interface{}, 0), "logs_info", progress),
+			"controller_logs_started_at", strfmt.DateTime(time.Now())), extra...)
+	default:
+		extra = append(append(make([]interface{}, 0), "logs_info", progress), extra...)
+	}
+
+	if cluster, err = UpdateCluster(log, db, clusterId, srcStatus, extra...); err != nil {
```

Comment:
use Model(c).Update instead of UpdateCluster to separate the update from the dependency on the state